### PR TITLE
added parsing for the last_accessed date arg

### PR DESCRIPTION
**Pull request by RynhardPietersen**
## Summary

a user was getting a 500 error on trying to undo removals for web users, and once you refresh the page, the web user is removed. Undo option was broken. It turns out the bug was with the last_accessed variable in the kwargs of add_domain_membership. I fixed this by just adding a parse for the variable.

## Product Description

Users can now remove web users. Whoo!

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below
It fixes a bug

### QA Plan

I tested the fix locally and will test it on staging.

### Safety story

Tested it locally. It worked! Will test on staging

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by biyeun — APPROVED** [review]

**Review by esoergel — CHANGES_REQUESTED** [review]
I don't think this is the right place to do date parsing.  This change would define `add_domain_membership` as expecting `last_accessed` to be a string representation of a date.  I'd instead expect callers of this method to provide the correct types already.  I'm guessing this call happens here, is that right?
https://github.com/dimagi/commcare-hq/blob/587339d84959036476067f74a75ad1173173028d/corehq/apps/users/models.py#L2675-L2682

`self.domain_membership` there is actually already defined as something matching the `DomainMembership` schema, though the JSON representation of this is passed in here.  `DomainMembership` (and any other Document or Jsonobject subclass) has a `wrap` method which is meant to accept JSON serialized objects.  I think it'd be safer to use that

```python
user.domain_memberships.append(
    DomainMembership.wrap(self.domain_membership._doc)
)
user.domains.append(domain)
```

Or hell, I can't think of any issues with just adding it directly

```python
user.domain_memberships.append(self.domain_membership)
user.domains.append(domain)
```

Aside: this code is really really old and looks pretty poor quality.  Like, what is this first check even doing?  "If you're adding a _new_ domain membership, and there's already an existing domain membership in there for that domain, make sure it's also already in `domains`."  Very odd place to check for that.

**Comment by RynhardPietersen** [comment]
@esoergel I took in to account your suggestions and changed the code.

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -2678,7 +2678,9 @@ class DomainRemovalRecord(DeleteRecord):
 
     def undo(self):
         user = WebUser.get_by_user_id(self.user_id)
-        user.add_domain_membership(**self.domain_membership._doc)
+        user.domain_memberships.append(self.domain_membership)
+        user.domains.append(self.domain)
+        #user.add_domain_membership(**self.domain_membership._doc)
```
Comment (esoergel) [inline corehq/apps/users/models.py:2683]:
```suggestion
```

**Review by esoergel — APPROVED** [review]
You left in a commented out line, otherwise approved

**Pull request merged**